crates/walrus-sui/src/client/retry_client/retriable_rpc_client.rs

```diff
@@ -103,14 +103,26 @@ pub enum LazyFallibleRpcClientBuilder {
 impl LazyClientBuilder<FallibleRpcClient> for LazyFallibleRpcClientBuilder {
     const DEFAULT_MAX_TRIES: usize = 5;
 
-    async fn lazy_build_client(
-        &self,
-        _is_last_chance: bool,
-    ) -> Result<Arc<FallibleRpcClient>, FailoverError> {
+    async fn lazy_build_client(&self) -> Result<Arc<FallibleRpcClient>, FailoverError> {
         // Inject rpc client build failure for simtests.
         #[cfg(msim)]
         {
-            if !_is_last_chance {
+            let mut fail_client_creation = false;
```

Comment:
I would like to change the failure injection back to prior #2052 because since now ContractClient created from wallet should be able to leverage multiple RPCs, the old test semantic makes sure that we do fail the primary RPCs. This is also the way to test the change made in this PR.